File: src/types/todo.ts
export interface Todo {
  id: number;
  title: string;
  description: string;
  priority: number; // 1=High, 2=Medium, 3=Low
  completed: boolean;
  deadline: string | null; // ISO 8601
  recurrence: Recurrence;
  recurrence_anchor: string | null;
  recurrence_weekday: number | null; // 1=Monday, 7=Sunday
  recurrence_month_day: number | null; // 1-31
  reminder_minutes_before: number | null;
  reminder_due_at: string | null;
  reminder_state: ReminderState;
  last_reminded_at: string | null;
  last_reminded_deadline: string | null;
  created_at: string;
}

export type Recurrence = "none" | "daily" | "weekly" | "monthly" | "yearly";
export type ReminderState = "none" | "upcoming" | "due" | "reminded" | "missed" | "overdue";

export interface CreateTodo {
  title: string;
  description?: string;
  priority?: number;
  deadline?: string;
  recurrence?: Recurrence;
  recurrence_weekday?: number;
  recurrence_month_day?: number;
  reminder_minutes_before?: number;
}

export interface UpdateTodo {
  id: number;
  title?: string;
  description?: string;
  priority?: number;
  deadline?: string;
  clear_deadline?: boolean;
  recurrence?: Recurrence;
  recurrence_weekday?: number;
  recurrence_month_day?: number;
  reminder_minutes_before?: number;
}

File: src-tauri/src/models/todo.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Todo {
    pub id: i64,
    pub title: String,
    pub description: String,
    pub priority: i32, // 1=High, 2=Medium, 3=Low
    pub completed: bool,
    pub deadline: Option<String>, // ISO 8601 format
    pub recurrence: String,       // none, daily, weekly, monthly, yearly
    pub recurrence_anchor: Option<String>,
    pub recurrence_weekday: Option<i64>,   // 1=Monday, 7=Sunday
    pub recurrence_month_day: Option<i64>, // 1-31
    pub reminder_minutes_before: Option<i64>,
    pub reminder_due_at: Option<String>,
    pub reminder_state: String, // none, upcoming, due, reminded, missed, overdue
    pub last_reminded_at: Option<String>,
    pub last_reminded_deadline: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateTodo {
    pub title: String,
    pub description: Option<String>,
    pub priority: Option<i32>,
    pub deadline: Option<String>,
    pub recurrence: Option<String>,
    pub recurrence_weekday: Option<i64>,
    pub recurrence_month_day: Option<i64>,
    pub reminder_minutes_before: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateTodo {
    pub id: i64,
    pub title: Option<String>,
    pub description: Option<String>,
    pub priority: Option<i32>,
    pub deadline: Option<String>,
    pub clear_deadline: Option<bool>,
    pub recurrence: Option<String>,
    pub recurrence_weekday: Option<i64>,
    pub recurrence_month_day: Option<i64>,
    pub reminder_minutes_before: Option<i64>,
}
